pandora-client-web/src/components/wardrobe/modules/wardrobeModuleLockSlot.tsx

```diff
@@ -306,38 +306,49 @@ function PasswordInput({
 
 	const id = useId();
 
+	let inputCharacterType: string;
+	let replaceFunc: ((_: string) => string);
+
+	switch (password.format) {
+		case 'numeric':
+			inputCharacterType = 'digits';
+			replaceFunc = (v) => v.replace(/[^0-9]/g, '');
```

Comment:
unfortunately moving this out of `useCallback` is not ideal as lambda functions will return a new object on each render so the `useCallback` will have no effect as `replaceFunc` change every time

Reply:
I did go for useMemo to memoize the function and the string.
- I'm assuming using a tuple is not a problem?
- A little bit later in the code there is another switch on password.format and allow regex tests. I could also create the allow functions when I create the replace functions. This way the code is next to each other.

Reply:
yeps this works, tuples are ok